### PR TITLE
Gen 1: Fix Transform bugs

### PR DESCRIPTION
**Pull request by Karthik99999**
- Fix copying stat boosts (https://www.smogon.com/forums/threads/bug-reports-v4-read-original-post-before-posting.3663703/post-8843431)
- Fix crit rate not using base species for speed stat (https://www.smogon.com/forums/threads/rby-tradebacks-bug-report-thread.3524844/post-6283457)

### CODE REVIEW

File: sim/pokemon.ts [hunk 1]
```diff
@@ -1174,7 +1174,15 @@ export class Pokemon {
 		}
 		let boostName: BoostID;
 		for (boostName in pokemon.boosts) {
-			this.boosts[boostName] = pokemon.boosts[boostName]!;
+			this.boosts[boostName] = pokemon.boosts[boostName];
+			if (this.battle.gen <= 1) {
```
Comment (DaWoblefet) [inline sim/pokemon.ts:1178]: I would just do equals gen 1, there's nothing prior to that.

Comment (Karthik99999) [reply sim/pokemon.ts:1178]: there was other code in this file that does `<= 1` which is why I did that (`=== 1` and `<= 1` are used pretty interchangeably throughout the entire codebase though so 🤷‍♂️)

Comment (DaWoblefet) [reply sim/pokemon.ts:1178]: Idk why we'd do it that way, but I suppose that can be a thing for another day.

**Comment by DaWoblefet** [comment]
Thanks as always, Karthik!

**Pull request merged**